### ISSUE TITLE
Super-linter returns false positives when the default branch is wrong

### ISSUE THREAD

**Issue by krdenison** [super-linter/super-linter#4541]
## Describe the bug

When `VALIDATE_ALL_CODEBASE: false` is set and the default branch parameter is wrong super linter returns success even though no files get linted. It throws a fatal error when trying to find which files have changed and then declares that nothing needs linting and returns green.

## Expected behavior

If an error happens during the linting process the job should return as failed.

## Steps to Reproduce

1. Setup super-linter on a git repository with `VALIDATE_ALL_CODEBASE: false` and set the `DEFAULT_BRANCH` to a branch that doesn't exist.
1. Open a PR with some changes and let the action run.
1. The action will return as successful/passed even though the job throws an error and doesn't lint any files.

## Logs

```
2023-08-04 19:19:05 [INFO]   User provided file:[/github/workspace/.github/linters/.yaml-lint.yml] exists, setting rules file...
You are not currently on a branch.
Please specify which branch you want to merge with.
See git-pull(1) for details.

    git pull <remote> <branch>

fatal: ambiguous argument 'master...bd0c69de34ffe7bd20feb3f9ec140d1a06819f5f': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
2023-08-04 19:19:06 [WARN]   No files were found in the GITHUB_WORKSPACE:[/github/workspace] to lint!
```
